apps/src/templates/progress/ProgressLegend.jsx

```diff
@@ -1,352 +1,325 @@
-                <ProgressBubble
-                  level={{
-                    id: '1',
-                    status: LevelStatus.perfect,
-                    isLocked: false,
-                    isConceptLevel: false,
-                    name: `${i18n.activity()}: ${i18n.completed()} (${i18n.perfect()})`
-                  }}
-                  disabled={false}
-                />
-              </div>
-            </TD>
-            <TD rowSpan={secondRowRowSpan}>
-              <div style={styles.center}>
-                <ProgressBubble
-                  level={{
-                    id: '1',
-                    status: LevelStatus.submitted,
-                    isLocked: false,
-                    isConceptLevel: false,
-                    name: `${i18n.activity()}: ${i18n.submitted()}`
-                  }}
-                  disabled={false}
-                />
-              </div>
-            </TD>
-          </tr>
+          <tr>{rows[1].map(cell => cell)}</tr>
+          <tr>{rows[2].map(cell => cell)}</tr>
         </tbody>
       </table>
     );
   }
-}
 
-const styles = {
-  table: {
-    textAlign: 'center',
-    width: '100%',
-    // Margin to get it to line up with ProgressLesson
-    marginLeft: 3,
-    marginRight: 3,
-    marginTop: 60
-  },
-  tdStyle: {
-    padding: 10,
-    borderStyle: 'none'
-  },
-  header: {
-    backgroundColor: color.white,
-    color: color.charcoal,
-    whiteSpace: 'nowrap'
-  },
-  secondRow: {
-    backgroundColor: color.lightest_gray,
-    color: color.charcoal,
-    borderWidth: 2,
-    borderColor: color.lightest_gray,
-    borderStyle: 'solid',
-    verticalAlign: 'top'
-  },
-  subsequentRow: {
-    backgroundColor: color.white,
-    borderWidth: 2,
-    borderColor: color.lightest_gray,
-    borderStyle: 'solid'
-  },
-  rightBorder: {
-    borderRightStyle: 'solid',
-    borderWidth: 2,
-    borderColor: color.lightest_gray
-  },
-  leftBorder: {
-    borderLeftStyle: 'solid',
-    borderWidth: 2,
-    borderColor: color.lightest_gray
-  },
-  headerCell: {
-    fontWeight: 'bold',
-    fontSize: 18
-  },
-  secondaryText: {
-    fontSize: 10
-  },
-  iconAndText: {
-    whiteSpace: 'nowrap'
-  },
-  iconAndTextDiv: {
-    whiteSpace: 'nowrap',
-    paddingBottom: 16
-  },
-  iconAndTextDivTop: {
-    whiteSpace: 'nowrap',
-    paddingTop: 10,
-    paddingBottom: 16
-  },
-  iconAndTextDivBottom: {
-    whiteSpace: 'nowrap',
-    paddingBottom: 10
-  },
-  icon: {
-    marginRight: 5,
-    size: 20
-  },
-  iconRTL: {
-    marginLeft: 5,
-    size: 20
-  },
-  center: {
-    display: 'flex',
-    alignItems: 'center',
-    justifyContent: 'center'
+  getLevelStatusColumns() {
+    const {
+      includeCsfColumn,
+      includeProgressNotApplicable,
+      includeReviewState
+    } = this.props;
+
+    const columns = [];
+    includeProgressNotApplicable && columns.push(this.getProgressNAColumn());
+    columns.push(this.getNotStartedColumn());
```

Comment:
Nit: with the lines where you're making 2 pushes in a row we could use concat to reduce to one line (I don't have a strong opinion on this it's just an option)